source/frontend/src/components/common/mapFSM/useMapSearch.tsx

```diff
@@ -0,0 +1,204 @@
+  emptyFeatureData,
+  emptyFullyFeaturedFeatureCollection,
+  emptyPimsFeatureCollection,
+  MapFeatureData,
+} from './models';
+
+export const useMapSearch = () => {
+  const fullyAttributedService = useFullyAttributedParcelMapLayer();
+  const pimsPropertyLayerService = usePimsPropertyLayer();
+
+  const { setModalContent, setDisplayModal } = useModalContext();
+  const keycloak = useKeycloakWrapper();
+  const logout = keycloak.obj.logout;
+
+  const pimsExecute = pimsPropertyLayerService.loadPropertyLayer.execute;
+  const fullyAttributedServiceFindByPin = fullyAttributedService.findByPin;
+  const fullyAttributedServiceFindByPid = fullyAttributedService.findByPid;
+
+  const fullyAttributedServiceFindOne = fullyAttributedService.findOne;
+  const pimsPropertyLayerServiceFindOne = pimsPropertyLayerService.findOne;
+
+  const searchOneLocation = useCallback(
+    async (latitude: number, longitude: number) => {
+      debugger;
+      let result: MapFeatureData = emptyFeatureData;
+      try {
+        const findOneParcelTask = fullyAttributedServiceFindOne({
+          lat: latitude,
+          lng: longitude,
+        });
+        const findOnePropertyTask = pimsPropertyLayerServiceFindOne(
+          {
+            lat: latitude,
+            lng: longitude,
+          },
+          'GEOMETRY',
+        );
+        const parcelFeature = await findOneParcelTask;
+        const pimsPropertyFeature = await findOnePropertyTask;
+
+        //if found in inventory return or else non inventory
+        const foundGeometry = pimsPropertyFeature
+          ? pimsPropertyFeature.geometry
+          : parcelFeature?.geometry;
+        if (pimsPropertyFeature !== undefined) {
+          result.pimsFeatures = { type: 'FeatureCollection', features: [pimsPropertyFeature] };
+        } else if (parcelFeature !== undefined) {
+          result.fullyAttributedFeatures = { type: 'FeatureCollection', features: [parcelFeature] };
+        }
+
+        if (foundGeometry !== undefined) {
+          if (pimsPropertyFeature !== undefined) {
+            result.pimsFeatures = { type: 'FeatureCollection', features: [pimsPropertyFeature] };
+          } else if (parcelFeature !== undefined) {
+            result.fullyAttributedFeatures = {
+              type: 'FeatureCollection',
+              features: [parcelFeature],
+            };
+          }
+          toast.info(`Property found`);
+        } else {
+          toast.info('No search results found');
+        }
+      } catch (error) {
+        toast.error((error as Error).message, { autoClose: 7000 });
+      } finally {
+      }
+      return result;
+    },
+    [fullyAttributedServiceFindOne, pimsPropertyLayerServiceFindOne],
+  );
+
+  const searchMany = useCallback(
+    async (filter?: IGeoSearchParams) => {
+      let result: MapFeatureData = emptyFeatureData;
+      //TODO: PSP-4390 currently this loads all matching properties, this should be rewritten to use the bbox and make one request per tile.
```

Comment:
is this comment still relevant in this context?

Reply:
I believe it is given the search is still looking for the entire map instead of using the bounds.